ngi_reports/utils/entities.py

```diff
@@ -155,260 +171,467 @@ def populate(self, log, organism_names, **kwargs):
+            log.error("The source for data for project {} is not LIMS.".format(project))
             raise BaseException
 
-        proj_details = proj.get('details',{})
+        proj_details = proj.get("details", {})
 
-        if 'aborted' in proj_details:
-            log.warn('Project {} was aborted, so not proceeding.'.format(project))
-            sys.exit('Project {} was aborted, stopping execution...'.format(project))
+        if "aborted" in proj_details:
+            log.warn("Project {} was aborted, so not proceeding.".format(project))
+            sys.exit("Project {} was aborted, stopping execution...".format(project))
 
         if not id_view:
-            self.ngi_id = proj.get('project_id')
+            self.ngi_id = proj.get("project_id")
 
         for date in self.dates:
             self.dates[date] = proj_details.get(date, None)
 
-        if proj.get('project_summary',{}).get('all_samples_sequenced'):
-            self.dates['all_samples_sequenced'] = proj.get('project_summary',{}).get('all_samples_sequenced')
-
-        self.contact = proj.get('order_details', {}).get('owner', {}).get('email', 'NA')
-        self.application = proj.get('application')
-        self.num_samples = proj.get('no_of_samples')
-        self.ngi_facility = 'Genomics {} Stockholm'.format(proj_details.get('type')) if proj_details.get('type') else None
-        self.reference['genome'] = None if proj.get('reference_genome') == 'other' else proj.get('reference_genome')
-        self.reference['organism'] = organism_names.get(self.reference['genome'], None)
-        self.user_ID = proj_details.get('customer_project_reference','')
-        self.num_lanes = proj_details.get('sequence_units_ordered_(lanes)')
-        self.library_construction_method = proj_details.get('library_construction_method')
-        self.library_prep_option = proj_details.get('library_prep_option', '')
-
-        if 'dds' in proj.get('delivery_type','').lower():
-            self.cluster = 'dds'
-        elif 'hdd' in proj.get('delivery_type','').lower():
-            self.cluster = 'hdd'
+        if proj.get("project_summary", {}).get("all_samples_sequenced"):
+            self.dates["all_samples_sequenced"] = proj.get("project_summary", {}).get(
+                "all_samples_sequenced"
+            )
+
+        self.contact = proj.get("order_details", {}).get("owner", {}).get("email", "NA")
+        self.application = proj.get("application")
+        if proj_details.get("sequencing_platform") in [
+            "MiSeq",
+            "NextSeq 2000",
+            "NovaSeq 6000",
+            "NovaSeq X Plus",
+        ]:
+            self.sequencer_manufacturer = "illumina"
+        elif proj_details.get("sequencing_platform") in ["PromethION", "MinION"]:
+            self.sequencer_manufacturer = "ont"
+        elif proj_details.get("sequencing_platform") in [
+            "Element AVITI"
+        ]:
+            self.sequencer_manufacturer = "element"
```

Comment:
What should be done about projects where the sequencing_platform is blank? Like P29904. From the running notes, it looks like its to be run on NovaSeqXPlus

Reply:
Hm... I guess that the best thing would be to add the info to statusdb 😅 In this case it was run several times on NovaSeqXPlus so I'm not sure why it's not filled in. Are single cell projects special? Could they be run on different sequencers in the same project? I guess I should at least add an `else` and produce a warning and possibly exit if this info is missing.

Reply:
An else with error message and exit seems good for now. We can update the code later if its something that becomes a problem too often.